R/data_match.R

```diff
@@ -152,6 +152,31 @@ data_filter <- function(x, filter, ...) {
   if (is.numeric(rows)) {
     out <- x[rows, , drop = FALSE]
   } else {
+
+    # check for curley tags, like "{var} == 1"
+    has_curley <- tryCatch(grepl("{", eval(condition), fixed = TRUE),
+                           error = function(e) FALSE)
```

Comment:
`eval(condition)` splits the "filter" argument at every `{` and `}`, resulting in a character vector length > 1, however we just want the `filter` argument as on string here.